python/pyspark/sql/column.py

```diff
@@ -906,7 +906,7 @@ def __nonzero__(self):
     __bool__ = __nonzero__
 
     def __repr__(self):
-        return 'Column<%s>' % self._jc.toString().encode('utf8')
+        return "Column<'%s'>" % self._jc.toString()
```

Comment:
python3 uses utf8 for strings by default, so this change seems fine. cc: @HyukjinKwon @viirya @srowen

Reply:
Do we have any more instances of decode()?

Reply:
Yeah, the change looks good, and yeah, are there more instances like this?

Reply:
Seems fine. Is it originally for non printable characters in unicode?

Reply:
> Do we have any more instances of decode()?

We do a bit of encoding / decoding when we communicate with JVM, but purpose there is clear.

The only other place when we encode strings intended for user consumption is `RDD.toDebugString`. It  also something that could be fixed, as it messing with the output a bit (as print won't respect line breaks).

> Seems fine. Is it originally for non printable characters in unicode?

I believe the point was to have `str` object as the output, instead of `unicode`. If I recall correctly, `unicode` (py4j returns JVM `Strings` as `unicode` in Python 2 and as a result the whole expression would evaluate to `unicode`)  in `__repr__`, wasn't handled correctly.

Reply:
For example in IPython

```
>>> import sys
>>> sys.version_info
sys.version_info(major=2, minor=7, micro=15, releaselevel='final', serial=0)
>>> class Foo:
...     def __repr__(self):
...         return u"œ"
...     
...     
>>> Foo().__repr__()
u'\u0153'
>>> Foo()
Traceback (most recent call last):
  File "/path/to/lib/python2.7/site-packages/IPython/core/formatters.py", line 686, in __call__
    return repr(obj)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u0153' in position 0: ordinal not in range(128)
```